tests/testthat/test-copy-model-from.R

```diff
@@ -213,3 +213,53 @@ test_that("copy_model_from(.new_model=NULL) errors when no models are valid inte
     new_mod2 <- copy_model_from(new_mod1)
   }, regexp = "no models.+integer")
 })
+
+test_that("copy_model_from works with various PROBLEM declarations [BBR-CMF-007]", {
+  withr::defer(cleanup())
+
+  get_prob_statement <- function(.mod){
+    ctl <- nmrec::read_ctl(get_model_path(.mod))
+    prob_rec <- nmrec::select_records(ctl, "prob")[[1]]
+    gsub("\n", "", prob_rec$format())
+  }
+
+  # Test lower case ($prob)
+  mod_content <- readLines(ctl_ext(MOD1_PATH))
+  mod_content[1] <- gsub("PROBLEM", "problem", mod_content[1])
+  mod_content <- mod_content %>% paste(collapse = "\n")
+  temp_mod_path <- create_temp_model(mod_content = mod_content)
+  temp_mod <- read_model(temp_mod_path)
+
+  new_mod_path <- "lowercase"
+  new_mod <- copy_model_from(temp_mod, new_mod_path, .overwrite = TRUE)
```

Comment:
I added `.overwrite = TRUE` here because subsequent runs would fail due to the existence of previously copied models. This shouldnt be needed, and I think this indicates a flaw in the `cleanup` function, but I couldn't narrow it down and figured we could address that in a separate PR if need be

Reply:
Okay, thanks for noting.